service/controller/clusterapi/v17/machine_deployment_resource_set.go

```diff
@@ -115,3 +119,19 @@ func NewMachineDeploymentResourceSet(config MachineDeploymentResourceSetConfig)
 
 	return resourceSet, nil
 }
+
+func newMachineDeploymentToClusterFunc(cmaClient clientset.Interface) func(obj interface{}) (v1alpha1.Cluster, error) {
```

Comment:
Couldn't find a better place for this.

Reply:
Why this can't be in service/controller/clusterapi/v17/resources/tenantclients/resource.go?

Reply:
It can, when we think this is a better place for it. I just didn't see that.

Reply:
I think that would allow to get rid of `Config.ToClusterFunc`?

Reply:
No, the transform function has to be configured because the resource is used in different controllers. When the `Cluster` watching controller executes the resource, it uses the key function. When the `MachineDeployment` watching controller executes the resource, it uses the lookup function here. When you want to get rid of the transform function we can revert the PR and have the type switch in the resource again. But that was way worse IMO.

Reply:
Ack. Then let's keep it here.